flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -269,6 +267,17 @@ private MemorySegment requestMemorySegmentFromGlobal() throws IOException {
 			if (segment != null) {
 				numberOfRequestedMemorySegments++;
 				return segment;
+			} else if (isBlocking) {
+				// if the future is completed before the callback is registered,
+				// the request thread will wait 2s before polling an available
+				// segment from the global pool, which is not a big problem.
+				networkBufferPool.isAvailable().thenRun(() -> {
+					if (!Thread.holdsLock(availableMemorySegments)) {
+						synchronized (availableMemorySegments) {
+							availableMemorySegments.notifyAll();
```

Comment:
Use availableMemorySegments.notify() instead? 
There was only one thread waiting and in `LocalBufferPool#recycle` it also use `notify`.